### ISSUE TITLE
Feature request to have an argument for auto answer for input prompts during scan.

### ISSUE THREAD

**Issue by rams3sh** [drwetter/testssl.sh#1419]
Please make sure that you provide enough information so that we understand what your issue is about.

1. uname -a
`Linux FDLDL064 5.0.0-37-generic #40~18.04.1-Ubuntu SMP Thu Nov 14 12:06:39 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`   
2. testssl version from the banner: testssl.sh -b 2>/dev/null | head -4 | tail -2
```
    testssl.sh       3.0rc6 from https://testssl.sh/dev/
    (fa5bb18 2019-12-24 11:47:41 -- )

```
3. git log | head -1 (if running from git repo)

`commit fa5bb18d15a4d6a88d9722feb393a56e8e763c3f`

4. openssl version used by testssl.sh: testssl.sh -b 2>/dev/null | awk -F':' '/openssl/ { print $2}'
`./bin/openssl.Linux.x86_64`
5. steps to reproduce: testssl.sh or docker command line, if possible incl. host
`./testssl.sh -t smtp   *****[MASKED]*****`

6. what exactly was happening, output is needed
In case -t is given as argument , if STARTTLS handshake fails , there is a prompt for providing an input. Output given below :- 
```
./testssl.sh -t smtp  *****[MASKED]*****

###########################################################
    testssl.sh       3.0rc6 from https://testssl.sh/dev/
    (fa5bb18 2019-12-24 11:47:41 -- )

      This program is free software. Distribution and
             modification under GPLv2 permitted.
      USAGE w/o ANY WARRANTY. USE IT AT YOUR OWN RISK!

       Please file bugs @ https://testssl.sh/bugs/

###########################################################

 Using "OpenSSL 1.0.2-chacha (1.0.2k-dev)" [~183 ciphers]
 on FDLDL064:./bin/openssl.Linux.x86_64
 (built: "Jan 18 17:12:17 2019", platform: "linux-x86_64")


 Start 2019-12-31 12:40:04        -->> *****[MASKED]***** <<--

 Further IP addresses:    *****[MASKED]*****
 rDNS ( *****[MASKED]*****):  *****[MASKED]*****
 STARTTLS handshake failed
 STARTTLS handshake failed
 STARTTLS handshake failed
 STARTTLS handshake failed
 STARTTLS handshake failed
 STARTTLS handshake failed
 STARTTLS handshake failed
 STARTTLS handshake failed

  *****[MASKED]***** doesn't seem to be a TLS/SSL enabled server
 The results might look ok but they could be nonsense. Really proceed ? ("yes" to continue) --> 

```

7. what did you expect instead?
In case of running for bulk list of  domains, it would not be possible for providing `yes` or `no` to each of the prompt in case the handshake fails. It would be nice to have an auto response as an argument something like `--auto-answer-prompt=[yes/no/default]` (where `yes` and `no` have their respective meanings and `default` stands for which ever decision is considered as a default for that prompt)  which will consider the same response for any prompt in case encountered during the scan for any check. This will be helpful for automation. If there is any other way currently in the tool which I may be ignorant about , which is being used for same purpose by others , request you to let me know or add it as part of `--help` as it may be helpful for others like me.

Thanks.

**Comment by dcooper16** [comment]
The closest option currently available for the feature you are requesting is `--warnings`:

> `--warnings <batch|off|false>` The warnings parameter determines how testssl.sh will deal with situations where user input normally will be necessary. There are a couple of options here. `batch` doesn't wait for a confirming keypress. This is automatically being chosen for mass testing (`--file`). `-false` just skips the warning AND the confirmation. Please note that there are conflicts where testssl.sh will still ask for confirmation which are the ones which otherwise would have a drastic impact on the results. Almost any other decision will be made as a best guess by testssl.sh. The same can be achieved by setting the environment variable `WARNINGS`.

The description seems to be a bit out of date. Only  the `batch` and `off` options are currently valid, and the behavior is the same whichever one you choose. As noted in the description, this option is automatically set if bulk testing is being performed (i.e., if the `--file` option is used).

The flag doesn't allow you to specify an answer to the prompt. If a question such as the one above is presented, setting `--warnings` to `batch` or `off` results in the question being answered with `no`.

**Comment by dcooper16** [comment]
Hi @drwetter,

As I noted above, the description of the `--warnings` flag is out of date in both the online documentation and in the `--help` output from testssl.sh. They indicate that `false` is an option, but testssl.sh will fail if the `--warnings` flag is provided on the command line with a value other than `batch` or `off`.

**Comment by drwetter** [comment]
@dcooper16 + wrt to ``--warnings=false``: That must have been a long time ago. It's not only that it'll fail, there's no such functionality.

**Comment by drwetter** [comment]
--> look closer at it next week. Likely there will be no re-implementation of the value ```false`` and not sure whether there will be a ``yes`` value

**Comment by drwetter** [comment]
TL;DR: false was obsoleted correctly and there doesn't seem to be a need for it.  @rams3sh : please try ``--warnings=off`` if you want to continue and  ``--warnings=batch`` if not. 


Previously (found it e.g. in  2.7dev) ``ignore_no_or_lame()``  was

```
ignore_no_or_lame() {
     local a

     [[ "$WARNINGS" == "off" ]] && return 0
     [[ "$WARNINGS" == "false" ]] && return 0
     [[ "$WARNINGS" == "batch" ]] && return 1
     pr_magenta "$1 "
     read a
     case $a in
          Y|y|Yes|YES|yes) return 0;;
          default)         ;;
     esac
     return 1
}
```

Now it checks whether $2 matches somehow the string interactively supplied. If it does it returns 0, if not 1. _batch_ and _off_ are different: _batch_ always returns 1, _off_ 0.

``grep -B2 ignore_no_or_lame testssl.sh`` shows however that in all cases testssl exits when ``ignore_no_or_lame`` returns not zero 

``[[ $? -ne 0 ]] && exit $ERR_RESOURCE|$ERR_CLUELESS)``

So in case of mass testing or _batch_ it exists when a condition is met and as a consequence ``ignore_no_or_lame()`` will be called. ``off`` testssl.sh always continues. This is equivalent to ``--auto-answer-prompt`` ``no`` and ``yes``, respectively as @rams3sh suggested. _default_ is missing but the the strings interactively supplied are always "yes" _default_ is equivalent to _batch_ , there's just no such thing as different answers.

**Comment by drwetter** [comment]
ping  @rams3sh. Have you tried? That should cover your usecase.

**Comment by drwetter** [comment]
timed out, autoclosing. Let me know if there's new input
